src/Parser/Core/Modules/DeathTracker.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import { formatMilliseconds } from 'common/format';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+const debug = true;
+
+class DeathTracker extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  deathTimestamp = 0;
+  resurrectTimestamp = 0;
+  timeDead = 0;
+  castCount = 0;
+  isAlive = true;
+
+  on_toPlayer_death(event) {
+    this.deathTimestamp = this.owner.currentTimestamp;
+    if (debug) {console.log("Player Died @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));}
+    this.isAlive = false;
+  }
+
+  on_toPlayer_resurrect(event) {
+    this.resurrectTimestamp = this.owner.currentTimestamp;
+    this.timeDead += this.resurrectTimestamp - this.deathTimestamp;
+    if (debug) {console.log("Player was Resurrected @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));}
+    this.isAlive = true;
+  }
+
+  on_byPlayer_cast(event) {
+    this.castCount += 1;
+  }
+
+  on_finished() {
+    if (this.isAlive === false) {
+      this.timeDead += this.owner.currentTimestamp - this.deathTimestamp;
+    }
+  }
+
```

Comment:
for less repeated code, could add in 
```
get timeDeadSeconds() {
   return (this.timeDead / 1000).toFixed(0);
}

get timeDeadPercent() {
   return this.timeDead / this.owner.fightDuration;
}
```

I've found `get` statements to be really nice for cleaning up code, as you can now use those in your suggestions without having to do the calculation each time, and it's still clear what you're doing.